akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala

```diff
@@ -128,6 +131,13 @@ abstract class BasicDirectives {
     D.extractExecutionContext { c ⇒ inner.apply(c).toScala })
 
   /**
+   * Extracts a single value using the given function.
+   */
+  def extract[T](extract: JFunction[RequestContext, T], inner: JFunction[T, Route]): Route = ScalaRoute {
```

Comment:
> are you actually planning to extract a whole Java requestcontext? I didn't do that, since it would've added a lot of Java work (that part is really scala-only).

As library authors we take all the pain ourselves such that users don't have to :wink:
It takes a special kind of masochism (and / or dedication) to care about the Java API being complete :wink: 

> I couldn't see any real use cases for it, even though it would of course be needed if you want 100% feature parity. It would also require introducing RouteResult on the Java side.

It is a pretty core concept and would be weird to skip now that I'm browsing things.

https://github.com/akka/akka/blob/master/akka-http/src/main/scala/akka/http/javadsl/server/RequestContext.scala

https://github.com/akka/akka/pull/20088/files#diff-fe1dc8b93e367e290481ad720726ef32L17